### PR TITLE
CSR Activities to Socrata

### PR DESCRIPTION
**Pull request by Charlie-Henry**
https://github.com/cityofaustin/atd-data-tech/issues/18595

ETL for CSR activities. Dataset is being loaded [here](https://datahub.austintexas.gov/Transportation-and-Mobility/311-Service-Request-Activities-Austin-Transportati/cqc3-f6xh/about_data).

I also did some cleaning up of my code to move some of the more repetitive stuff into `etl/utils.py`

## Testing
If you would like to test this (idk if necessary, we can wait until I have something set up in airflow if you want) , I have provided an env file in the DTS dev vault called `dts-311-reporting Env file for testing`. 

To test, make sure you're on the city network and just run `$ python etl/activities_to_socrata.py` and check if you get a response from socrata something like:

> 2024-08-12 15:31:55,570 INFO: {'Errors': 0, 'Rows Created': 0, 'Rows Updated': 394133, 'Rows Deleted': 0}

### CODE REVIEW

File: etl/activities_to_socrata.py
Comment (johnclary) [inline etl/activities_to_socrata.py:1]: this file might need a shebang `#!`

File: etl/activities_to_socrata.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Field mapping
+    df = df[list(ACTIVITIES_MAP.keys())]
+    df.rename(columns=ACTIVITIES_MAP, inplace=True)
+
+    # date column formatting to match format expected by Socrata
+    df = utils.transform_datetime_formats(df)
+
+    # Remove rows without an activity id, this happens when a CSR has no activities associated with it
+    df = df.dropna(subset=["activity_id"])
```
Comment (johnclary) [inline etl/activities_to_socrata.py:36]: i am having trouble picturing this—you have rows of activities and some don't have IDs? as if they are placeholder/null activities?

**Review by johnclary — APPROVED** [review]
looks good to me—very nice to see you extending this repo to handle another dataset 🙌

File: etl/activities_to_socrata.py (hunk 1 above)
Comment (Charlie-Henry) [reply etl/activities_to_socrata.py:36]: @johnclary, this is that the report looks like. Just a bunch of SR numbers and no activity ID/activity data. If you look these SRs up, you'll see they all have zero activities so I think this is a quirk of how 311 set up this report for us.
 
Screenshot 2024-08-13 at 12 10 13 PM

Comment (johnclary) [reply etl/activities_to_socrata.py:36]: 🙃 i see now thank youu

**Review by susannegov — APPROVED** [review]
Went over it with Charlie in Data Science Backlog refinement. Looks good!

**Review by ChristinaTremel — APPROVED** [review]
Looks goooood! 🚀

**Comment by Charlie-Henry** [comment]
Thanks all!

**Pull request merged**